### PR TITLE
Make MariaDB CRC32-lib platform independence

### PR DESCRIPTION
**Pull request by guyuqi**
It looks strange that someone can make use of 2 crc libraries(Power64 or AArch64)at the same time.
The patch set macros 'CRC32_LIBRARY' to make platform independence as a optional crc32 library.

In PR #772,  I'll add Arm64 crc32 support to CRC32_LIBRARY in 'crc32.cmake' .

### CODE REVIEW

**Comment by grooverdan** [comment]
tested on ppc64le, POWER8 :+1:

File: CMakeLists.txt [hunk 1]
```diff
@@ -166,7 +166,7 @@ INCLUDE(install_macros)
 INCLUDE(systemd)
 INCLUDE(mysql_add_executable)
 INCLUDE(compile_flags)
-INCLUDE(crc32-vpmsum)
+INCLUDE(crc32_boost)
```
Comment (svoj) [inline CMakeLists.txt]: I'd prefer just `crc32`.

File: cmake/crc32_boost.cmake [hunk 2]
```diff
@@ -2,4 +2,5 @@ IF(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64")
     SET(HAVE_CRC32_VPMSUM 1)
     SET(CRC32_VPMSUM_LIBRARY crc32-vpmsum)
     ADD_SUBDIRECTORY(extra/crc32-vpmsum)
+    SET(CRC32_BOOST_LIBRARY ${CRC32_VPMSUM_LIBRARY})
```
Comment (svoj) [inline cmake/crc32_boost.cmake]: I believe we should remove `CRC32_VPMSUM_LIBRARY` and use `crc32-vpmsum` directly.

Comment (svoj) [reply cmake/crc32_boost.cmake]: Also please use `CRC32_LIBRARY` not to confuse with well known boost.

Comment (guyuqi) [reply cmake/crc32_boost.cmake]: updated

**Comment by svoj** [comment]
Hi @guyuqi,

Thanks for your contribution.

Similar to other open source projects, the MariaDB Foundation needs to have shared ownership of all code that is included in the MariaDB distribution. The easiest way to achieve this is by submitting your code under the BSD-new license.

The other alternative is to sign the code contribution agreement which can be found here: https://mariadb.com/kb/en/mariadb/mca/

Please indicate in a comment below that you are contributing your new code of the whole pull request, including one or several files that are either new files or modified ones, under the BSD-new license or that you have filled out the contribution agreement and sent it.

Thanks,
Sergey

**Comment by guyuqi** [comment]
Hi @svoj ,

I also found mca on:
https://mariadb.org/get-involved/getting-started-for-developers/mca/

It's noted that there is little difference in "Signatory Information":
"Your user name on Launchpad / askmonty.org:"
and "Your user name on GitHub:".

Could you please tell me which one should I choose? 
Thanks!

BRs,
Yuqi

**Comment by svoj** [comment]
@guyuqi my recommendation would be to use BSD-new. You just need to add a comment with phrase 
> I'm contributing this new code of the whole pull request, including one or several files that are either new files or modified ones, under the BSD-new license.

**Comment by guyuqi** [comment]
@svoj , thanks for your kindly suggestion. But in consideration of legal concern, I prefer to use mca.
I'll fille out the contribution agreement and sent it.  :-)

**Comment by svoj** [comment]
@guyuqi then please use MCA from mariadb.com. We don't use launchpad anymore.

**Comment by guyuqi** [comment]
I've signed MCA and sent it to foundation@mariadb.org.
I submit this PR under the MCA. Thanks!

**Comment by svoj** [comment]
Merged to 10.3 since the patch is fairly simple and there're more conflicting patches pending merge from 10.3 to 10.4.

Feel free to adjust #772 on top of 10.3, we'll cherry-pick it to 10.4 when merge is done.

**Comment by guyuqi** [comment]
@svoj, yes I will.

**Pull request merged**